core/src/main/java/org/mule/construct/AbstractPipeline.java

```diff
@@ -199,7 +201,7 @@ protected void doInitialise() throws MuleException
         if (messageSource != null)
         {
             // Wrap chain to decouple lifecycle
-            messageSource.setListener(new AbstractInterceptingMessageProcessor()
+            messageSource.setListener(new MessageProcessor()
```

Comment:
AbstractInterceptingMessageProcessor is not needed here as far as i can see.  If we change to MessageProcessor then non blocking is simplicity supported, ese it would need to implement NonBlockingSupported.